uiowa/uiowa
# Role-based content access per content type

Closes #3723 
Closes #3783 


# How to test



Will need to check multiple different sites (v2 default, v2 custom, v3 default, v3 custom) and make sure the users can “view own content” & “view any content” - if custom site, make sure available for all custom content types. If perms are diff for Page/Person/Article (for custom sites) that's because it's a one off. 

Admissions
User roles: authenticated, editor, publisher, webmaster (anon, viewer)

Brand
User roles: collaborator, editor, publisher, webmaster

V2 default
sitenow_v2 
User roles: editor, webmaster

Grad
User roles: anonymous, authenticated, editor, webmaster

HR
User roles: editor, publisher, webmaster

IGHN
User roles: editor, publisher, webmaster

policy_clas
User roles: editor, publisher, webmaster

Presidential search
User role: webmaster

SPPA
User role: authenticated

Transportation
User role: webmaster

## Site examples:
Before you sync a site, please run `composer install`

### V2 default:
`blt ds —site=hawkeyemarchingband.uiowa.edu`
`drush @hawkeyemarchingband.local uli`
https://hawkeyemarchingband.local.drupal.uiowa.edu/admin/people/permissions
Search for “Node view permissions” - should be set for editor and webmaster

### V2 custom:
`blt ds —site=hr.uiowa.edu`
`drush @hr.local uli`
https://hr.local.drupal.uiowa.edu/admin/people/permissions
Search for “Node view permissions” - Should be set for editor, publisher, webmaster

`blt ds —site=brand.uiowa.edu`
`drush @brand.local uli`
https://brand.local.drupal.uiowa.edu/admin/people/permissions
Search for “Node view permissions” - Should be set for collaborator, editor, publisher and webmaster
(Default content types should have the anonymous and authenticated perms set) 

### V3 default:
`blt ds —site=sandbox.uiowa.edu`
`drush @sandbox.local uli`
https://sandbox.local.drupal.uiowa.edu/admin/people/permissions
Search for “Node view permissions” - Should be set for editor, publisher and webmaster

### V3 custom:
`blt ds —site=admissions.uiowa.edu`
`drush @admissions.local uli`
https://admissions.local.drupal.uiowa.edu/admin/people/permissions
Search for “Node view permissions” - Should be set for anonymous and authenticated

`blt ds --site=sppa.uiowa.edu`
`drush @sppa.local uli`
https://sppa.local.drupal.uiowa.edu/admin/people/permissions
Search for “Node view permissions” - Opportunity should be set w/ authenticated perms
